src/wallet/wallet_rpc_server.cpp

```diff
@@ -1194,6 +1194,29 @@ namespace tools
     return true;
   }
   //------------------------------------------------------------------------------------------------------------------------------
+  bool wallet_rpc_server::on_set_wallet_note(const wallet_rpc::COMMAND_RPC_SET_WALLET_NOTE::request& req, wallet_rpc::COMMAND_RPC_SET_WALLET_NOTE::response& res, epee::json_rpc::error& er)
+  {
+    if (!m_wallet) return not_open(er);
+    if (m_wallet->restricted())
+    {
+      er.code = WALLET_RPC_ERROR_CODE_DENIED;
+      er.message = "Command unavailable in restricted mode.";
+      return false;
+    }
+
+    m_wallet->set_wallet_note(req.note);
+
+    return true;
+  }
+  //------------------------------------------------------------------------------------------------------------------------------
+  bool wallet_rpc_server::on_get_wallet_note(const wallet_rpc::COMMAND_RPC_GET_WALLET_NOTE::request& req, wallet_rpc::COMMAND_RPC_GET_WALLET_NOTE::response& res, epee::json_rpc::error& er)
+  {
+    if (!m_wallet) return not_open(er);
+
+    res.note = m_wallet->get_wallet_note();
+    return true;
+  }
+  //------------------------------------------------------------------------------------------------------------------------------
```

Comment:
This one is missing the restricted flag, and RPC should be generic (ie, pass the key you want to set/query).

Reply:
What's the "restricted flag"?

> pass the key you want to set/query

You are right, in the end one would probably like the whole key/value pair storing mechanism exposed, but I wanted to start small and restrict the whole thing to the wallet note for this first pull request. Maybe anyway somebody with a little more knowledge than me would be needed to properly expose this through RPC...

Is this a question of keeping the RPC interface as small and generic as possible, so a special call just for the note is not a good idea?

Reply:
The restricted flag is a flag/mode/setting which prevents some calls that are deemed not to be exposed to third party users. I'd expect reading key/values is one of those, as it may contain private data.

About making the RPC more generic, I can do that on top of your patch if you prefer.